src/content/1.7/modules/sample-modules/extending-sf-form-with-upload-image-field.md

```diff
@@ -0,0 +1,586 @@
+---
+title: Extending Symfony form with upload image field
+weight: 3
+aliases: 
+    - /1.7/modules/sample_modules/grid-and-identifiable-object-form-hooks-usage
+---
+
+# Extending Symfony form with upload image field
+{{< minver v="1.7.7" title="true" >}}
+
+
+## Introduction
+
+In this tutorial we are going to build a module which extends `Suppliers` form 
+(SELL -> Catalog -> Brands & Suppliers). You will learn how to create:
```

Comment:
```suggestion
(SELL -> Catalog -> Brands & Suppliers). This module will address the following need "I'd like to add a 'upload image' field to the Supplier Add/Edit form because I want to display a new logo for each supplier on my shop. So this new field must allow me to add, edit and delete image files linked to a supplier. Obviously I expect the uploaded files to be stored in a consistent way with how PrestaShop usually stores such files."

We are going to create the module to address this need by using PrestaShop hooks in the Add/Edit Supplier Back-Office page and we will follow some software development best practices such as SOLID principles to make our code as clear and maintainable as possible.
You will learn how to create:
```

I'm not sure about the "we will follow some software development best practices such as SOLID principles to make our code as clear and maintainable as possible." part, what is your opinion @tdavidsonas88 ?

Reply:
SOLID practices are good as long as they are for the good purpose. If you are not sure then I'm double not sure @matks !

Reply:
I think as maintainable is enough and clear is not needed !

Reply:
Supported would be the best !